internal/services/rate.go

```diff
@@ -23,18 +23,26 @@ type Rate struct {
 
 var ErrNoFetchOccurred = errors.New("no fecth occurred yet")
 
+type ExchangeRateClient interface {
+	GetExchangeRate() (float32, error)
+}
+
 type CachingRateService struct {
-	currentRate    *Rate
+	currentRate    float32
 	mutex          sync.RWMutex
 	fetchError     error
 	updateInterval time.Duration
+	client         ExchangeRateClient
 }
 
-func NewRateService(updateInterval time.Duration) *CachingRateService {
+// Creates new rate service instance.
+// Pass nil to client to use default http client.
```

Comment:
tiny: For such cases [options pattern](https://dev.to/kittipat1413/understanding-the-options-pattern-in-go-390c) is used in Go

Reply:
Thanks a lot! Will use it in the next home task.